Python/Automate_Commiting_using_Git/automate_commiting_using_git.py

```diff
@@ -1,7 +1,42 @@
 import subprocess
 
-subprocess.getoutput("git add .")
-message = input("Enter commit message")
-subprocess.getoutput("git commit -m " + message)
-branchname = input("Enter branch name")
-subprocess.getoutput("git push origin " + branchname)
+# Stage all changes from the root of the git project
+result = subprocess.run(["git", "add", ":/"])
+if result.returncode != 0:
+    print("Error: Failed to stage changes.")
+    sys.exit(1)
+
+# Get commit message from user input or use default value
+message = input("Enter commit message (or press Enter to use default): ")
+if not message:
+    message = "Auto commit"
+
+# Get remote name from user input or use default value
+remote = input("Enter remote name (or press Enter to use default 'origin'): ")
+if not remote:
+    remote = "origin"
+    
+# Get current branch name
+result = subprocess.run(["git", "rev-parse", "--abbrev-ref", "HEAD"], capture_output=True, text=True)
+if result.returncode != 0:
+    print("Error: Failed to get current branch name.")
+    sys.exit(1)
+    
+branchname = input(f"Enter branch name (or press Enter to use default '{result.stdout.strip()}'): ")
+if not branchname:
+    branchname = result.stdout.strip()
```

Comment:
It would be enough to set `branchname = 'HEAD` per default. The rev-parse above is just an unnecessary step.

Reply:
Done